File: inove4us/backend/version_info.py
"""Versão de release do inove4us — lida de VERSION (+ GIT_SHA no deploy)."""

from __future__ import annotations

import os
from functools import lru_cache
from pathlib import Path

_ROOT = Path(__file__).resolve().parents[1]
_VERSION_FILE = _ROOT / "VERSION"


@lru_cache(maxsize=1)
def get_version() -> str:
    env = (os.environ.get("APP_VERSION") or os.environ.get("INOVE4US_VERSION") or "").strip()
    if env:
        return env
    try:
        return _VERSION_FILE.read_text(encoding="utf-8").strip() or "0.0.0"
    except OSError:
        return "0.0.0"


def get_git_sha() -> str:
    for key in ("GIT_SHA", "SOURCE_COMMIT", "GITHUB_SHA", "COMMIT_SHA"):
        val = (os.environ.get(key) or "").strip()
        if val:
            return val[:12]
    sha_file = _ROOT / "GIT_SHA"
    try:
        return sha_file.read_text(encoding="utf-8").strip()[:12] or "unknown"
    except OSError:
        return "unknown"


def version_payload(**extra) -> dict:
    payload = {
        "app": "inove4us",
        "version": get_version(),
        "git_sha": get_git_sha(),
    }
    payload.update(extra)
    return payload

File: leaction-platform/services/gateway-api/version-info.js
'use strict';

/**
 * Versão de release do Action Hub — lida de VERSION na raiz do leaction-platform.
 */

const fs = require('fs');
const path = require('path');

const ROOT = path.resolve(__dirname, '../..');
const VERSION_FILE = path.join(ROOT, 'VERSION');
const SHA_FILE = path.join(ROOT, 'GIT_SHA');

function getVersion() {
  const fromEnv = String(
    process.env.APP_VERSION || process.env.ACTIONHUB_VERSION || ''
  ).trim();
  if (fromEnv) return fromEnv;
  try {
    return fs.readFileSync(VERSION_FILE, 'utf8').trim() || '0.0.0';
  } catch {
    return '0.0.0';
  }
}

function getGitSha() {
  for (const key of ['GIT_SHA', 'SOURCE_COMMIT', 'GITHUB_SHA', 'COMMIT_SHA']) {
    const val = String(process.env[key] || '').trim();
    if (val) return val.slice(0, 12);
  }
  try {
    return fs.readFileSync(SHA_FILE, 'utf8').trim().slice(0, 12) || 'unknown';
  } catch {
    return 'unknown';
  }
}

function versionPayload(extra = {}) {
  return {
    app: 'actionhub',
    version: getVersion(),
    git_sha: getGitSha(),
    ...extra,
  };
}

module.exports = {
  getVersion,
  getGitSha,
  versionPayload,
};
